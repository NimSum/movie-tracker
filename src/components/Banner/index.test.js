import React from 'react';
import { shallow } from 'enzyme';
import { Banner, mapStateToProps } from './index';

describe('Banner', () => {
  const mockMovieUrls = Array(20).fill('https://image.tmdb.org/t/p/w500/movieposter.jpg/');
  let wrapper  = shallow(
      < Banner 
        posters={ mockMovieUrls } />
    )
  
    it.skip('should match snapshot with posters urls passed in', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it.skip('should match snapshot of show favorites', () => {
      wrapper  = shallow(
        < Banner 
          posters={ mockMovieUrls }
          showFavs={ true } />
      )
      expect(wrapper).toMatchSnapshot();
    })
    
    describe('mapStateToProps', () => {
      const mockState = {
        movies: Array(20).fill({ poster_path: 'movieposter.jpg/' }),
        showFavorites: false
      }
      const expected = {
        posters: mockMovieUrls,
        showFavs: false
      }
      expect(mapStateToProps(mockState)).toEqual(expected);
    })
})