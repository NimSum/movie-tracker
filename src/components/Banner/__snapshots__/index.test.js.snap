// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Should match snapshot with posters urls passed in 1`] = `ShallowWrapper {}`;
